File: DownloadLib/src/main/java/com/kronos/download/adapter/BaseObserveAdapter.java
package com.kronos.download.adapter;

/**
 * Created by Leif Zhang on 16/8/18.
 * Email leifzhanggithub@gmail.com
 */
public class BaseObserveAdapter implements IModelObservable {
    private final DataObservable mDataSetObservable;

    public BaseObserveAdapter() {
        mDataSetObservable = new DataObservable();
    }

    public void registerDataSetObserver(IObserver observer) {
        mDataSetObservable.registerObserver(observer);
    }

    public void unregisterDataSetObserver(IObserver observer) {
        mDataSetObservable.unregisterObserver(observer);
    }

    @Override
    public void notifyDataChange() {
        mDataSetObservable.notifyChanged();
    }

    @Override
    public void unregisterAll() {
        mDataSetObservable.unregisterAll();
    }

}

File: DownloadLib/src/main/java/com/kronos/download/adapter/BaseObserveAdapter.kt
package com.kronos.download.adapter

/**
 * Created by Leif Zhang on 16/8/18.
 * Email leifzhanggithub@gmail.com
 */
open class BaseObserveAdapter : IModelObservable {
    private val mDataSetObservable: DataObservable = DataObservable()

    override fun registerDataSetObserver(observer: IObserver) {
        mDataSetObservable.registerObserver(observer)
    }

    override fun unregisterDataSetObserver(observer: IObserver) {
        mDataSetObservable.unregisterObserver(observer)
    }

    override fun notifyDataChange() {
        mDataSetObservable.notifyChanged()
    }

    override fun unregisterAll() {
        mDataSetObservable.unregisterAll()
    }

}
